pipeline/clinvar/enigma_from_clinvar.py

```diff
@@ -0,0 +1,291 @@
+
+    if omim_name_el:
+        id_value = clinvar.textIfPresent(omim_name_el[0], "ElementValue")
+        condition_id_value = "{}; {} ({})".format(id_value, condition_id_symbol,
+                                                  condition_id_id)
+    else:
+        condition_id_value = default_val
+
+    return condition_id_type, condition_id_value
+
+
+def _extract_clinvar_accession(engima_assertion):
+    clinvar_accession_el = engima_assertion.find("ClinVarAccession")
+
+    if clinvar_accession_el is not None:
+        acc = clinvar_accession_el.get("Acc")
+        v = clinvar_accession_el.get("Version")
+
+        if acc is not None:
+            if v is not None:
+                return "{}.{}".format(acc, v)
+            else:
+                return acc
+
+    return default_val
+
+
+def _xpath(el, xpath):
+    e = el.xpath(xpath)
+    if e:
+        return e[0]
+    return default_val
+
+
+def _xpath_text(el, xpath):
+    e = _xpath(el, xpath)
+    if e is not None:
+        return e.text
+    return default_val
+
+
+# clinvar set element
+def parse_record(cvs_el, hgvs_util, assembly="GRCh38"):
+    '''
+    Extracts information out of a ClinVarSet XML element
+
+    :param cvs_el: ClinVarSet XML element
+    :param hgvs_util: instance of HGVS util to calculate protein changes
+    :param assembly: str assembly to use
+    :return: dictionary containing the extracted data. keys correspond to column names
+    '''
+    rec = {}
+
+    # find enigma assertion
+    enigma_assertion = cvs_el.xpath(
+        'ClinVarAssertion[contains(ClinVarSubmissionID/@submitter, "ENIGMA")]')[
+        0]
+
+    rec["Gene_symbol"] = _xpath_text(cvs_el,
+                                     'ReferenceClinVarAssertion/MeasureSet/Measure/MeasureRelationship/Symbol/ElementValue[starts-with(., "BRCA") and @Type="Preferred"]')
+
+    rec["Genomic_Coordinate"] = _extract_genomic_coordinates(cvs_el, assembly)
+
+    # getting hgvs cdna
+    hgvs_list = [e.text for e in enigma_assertion.findall(
+        'MeasureSet/Measure/AttributeSet/Attribute[@Type="HGVS"]')]
+
+    hgvs_list_filtered = [s for s in hgvs_list if re.match('NM_.*:.*', s)]
+
+    if hgvs_list_filtered:
+        hgvs_cdna_complete = hgvs_list_filtered[0]
+        rec["Reference_sequence"] = hgvs_cdna_complete.split(":")[0]
+        rec["HGVS_cDNA"] = hgvs_cdna_complete.split(":")[1]
+    else:
+        hgvs_cdna_complete = None
+        rec["Reference_sequence"] = default_val
+        rec["HGVS_cDNA"] = default_val
+
+    rec["BIC_Nomenclature"] = _fetch_bic(cvs_el)
+
+    rec["Abbrev_AA_change"], rec["HGVS_protein"] = _compute_protein_changes(
+        hgvs_cdna_complete, hgvs_util)
+
+    rec["URL"] = clinvar.textIfPresent(enigma_assertion,
+                                       'ClinicalSignificance/Citation/URL')
+
+    rec["Condition_ID_type"], rec[
+        "Condition_ID_value"] = _extract_condition_info(cvs_el)
+
+    trait_set_el = cvs_el.find('ReferenceClinVarAssertion/TraitSet')
```

Comment:
would`.findall()` be useful here?

Reply:
no, TraitSet appears precisely onceshttps://github.com/ncbi/clinvar/blob/master/xsd/clinvar_public.xsd#L300